.gitlab/llnl-ci-pip.yml

```diff
@@ -102,30 +67,60 @@ pip-tests:
     - python3 ci/tests/test.py -e
     - deactivate
 
-.pip-env-cleanup: &pip-env-cleanup
-  stage: cleanup
-  tags:
-    - ruby
-    - shell
-  script:
-    - source ${PIP_WORK_DIR}/ruby-env/bin/activate
-    - pip cache purge
-    - *finalize
-
-pip-cleanup-on-build-failure:
-  needs:
-    - job: pip-build
-  when: on_failure
-  <<: *pip-env-cleanup
-
-pip-cleanup-on-test-failure:
-  needs:
-    - job: pip-tests
-  when: on_failure
-  <<: *pip-env-cleanup
-
-pip-cleanup-on-success:
-  needs:
-    - job: pip-tests
-  when: on_success
-  <<: *pip-env-cleanup
+### JOBS
+# QUARTZ
+pip_setup_quartz:
+  variables:
+    RUNNER_TYPE: "shell"
+  extends: [.on_quartz, .job_tags, .pip_env_setup]
+
+pip_build_quartz:
+  variables:
+    RUNNER_TYPE: "shell"
+  needs: [pip_setup_quartz]
+  extends: [.on_quartz, .job_tags, .pip_build]
+
+pip_test_quartz:
+  variables:
+    RUNNER_TYPE: ["batch", "slurm"]
+    LLNL_SLURM_SCHEDULER_PARAMETERS: "--nodes=1 -t 30"
+  needs: [pip_build_quartz]
+  extends: [.on_quartz, .job_tags, .pip_test]
+
+# RUBY
+pip_setup_ruby:
+  variables:
+    RUNNER_TYPE: "shell"
+  extends: [.on_ruby, .job_tags, .pip_env_setup]
+
+pip_build_ruby:
+  variables:
+    RUNNER_TYPE: "shell"
+  needs: [pip_setup_ruby]
+  extends: [.on_ruby, .job_tags, .pip_build]
+
+pip_test_ruby:
+  variables:
+    RUNNER_TYPE: "batch"
+    LLNL_SLURM_SCHEDULER_PARAMETERS: "--nodes=1 -t 30"
+  needs: [pip_build_ruby]
+  extends: [.on_ruby, .job_tags, .pip_test]
+
+# LASSEN
+pip_setup_lassen:
+  variables:
+    RUNNER_TYPE: "shell"
+  extends: [.on_lassen, .job_tags, .pip_env_setup]
+
+pip_build_lassen:
+  variables:
+    RUNNER_TYPE: "shell"
+  needs: [pip_setup_lassen]
+  extends: [.on_lassen, .job_tags, .pip_build]
+
+pip_test_lassen:
+  variables:
+    RUNNER_TYPE: "batch"
+    LLNL_SLURM_SCHEDULER_PARAMETERS: "--nodes=1 -t 30"
```

Comment:
Lassen doesn't have this variable `LLNL_SLURM_SCHEDULER_PARAMETERS`, but have the following ones
```
LLNL_LSF_SCHEDULER_PARAMETERS: "-nnodes 1 -W 30"
LSB_JOB_STARTER: "ENVIRONMENT=BATCH /usr/tcetmp/bin/bsub_job_starter %USRCMD"
ENVIRONMENT: "BATCH"
```

Reply:
Copy-pasting issues, thanks for catching